src/components/flowchart-primary-toolbar/flowchart-primary-toolbar.js

```diff
@@ -91,6 +115,10 @@ export const mapDispatchToProps = (dispatch) => ({
   onToggleTextLabels: (value) => {
     dispatch(toggleTextLabels(Boolean(value)));
   },
+  onToggleExpandAllPipelines: (isExpanded) => {
+    dispatch(changeFlag('expandAllPipelines', isExpanded));
+    dispatch(loadInitialPipelineData());
```

Comment:
@SajidAlamQB Any reason why you club `changeFlag` & `loadInitialPipelineData` actions inside `onToggleExpandAllPipelines` and did not create a separate function for each action ?

Reply:
The `changeFlag` didn't update the UI reactively until you refreshed the page. From what I understand the `changeFlag` action modifies the state to reflect the new setting for expanding or collapsing all pipelines, but this change does not automatically trigger a refresh of the pipeline. By including `loadInitialPipelineData`, it reloaded the pipeline data whenever the flag changes.

Reply:
Make sense.